src/express/common/config-utils.js

```diff
@@ -44,63 +61,128 @@ const validCertPathStr = makeValidator((x) => {
+          'us-east-2',
+          'us-east-1',
+          'us-west-1',
+          'us-west-2',
+          'af-south-1',
+          'ap-east-1',
+          'ap-south-2',
+          'ap-southeast-3',
+          'ap-south-1',
+          'ap-northeast-3',
+          'ap-northeast-2',
+          'ap-southeast-1',
+          'ap-southeast-2',
+          'ap-northeast-1',
+          'ca-central-1',
+          'eu-central-1',
+          'eu-west-1',
+          'eu-west-2',
+          'eu-south-1',
+          'eu-west-3',
+          'eu-south-2',
+          'eu-north-1',
+          'eu-central-2',
+          'me-south-1',
+          'me-central-1',
+          'sa-east-1',
+          'us-gov-east-1',
+          'us-gov-west-1'
+        ],
+        docs:
+          'https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/' +
+          'Concepts.RegionsAndAvailabilityZones.html'
+      }),
+      PEM_FILE_PATH: validCertPathStr({ default: '/home/ubuntu/aws-key.pem' })
+    })
+
+  // validating GCP infra vars
+  } else if (cloud == 'gcp') {
+    cleanEnv(process.env, {
+      TF_VAR_MACHINE_TYPE: validStr({ default: 'n2d-standard-4' }),
+      TF_VAR_BOR_INSTANCE_TYPE: validStr({ default: 'n2d-standard-4' }),
+      TF_VAR_ERIGON_INSTANCE_TYPE: validStr({ default: 'n2d-standard-4' }),
+      TF_VAR_BOR_ARCHIVE_INSTANCE_TYPE: validStr({ default: 'n2d-standard-4' }),
+      TF_VAR_ERIGON_ARCHIVE_INSTANCE_TYPE: validStr({ default: 'n2d-standard-4' }),
+      TF_VAR_INSTANCE_IMAGE: validGCPVmImageStr({ default: 'ubuntu-2204-jammy-v20230302' }),
+      TF_VAR_REGION: validStr({
+        default: 'us-central1',
+        choices: [
+          "asia-east1",
+          "asia-east2",
+          "asia-northeast1",
+          "asia-northeast2",
+          "asia-northeast3",
+          "asia-south1",
+          "asia-south2",
+          "asia-southeast1",
+          "asia-southeast2",
+          "australia-southeast1",
+          "australia-southeast2",
+          "europe-central2",
+          "europe-north1",
+          "europe-southwest1",
+          "europe-west1",
+          "europe-west2",
+          "europe-west3",
+          "europe-west4",
+          "europe-west6",
+          "europe-west8",
+          "europe-west9",
+          "me-west1",
+          "northamerica-northeast1",
+          "northamerica-northeast2",
+          "southamerica-east1",
+          "southamerica-west1",
+          "us-central1",
+          "us-east1",
+          "us-east4",
+          "us-east5",
+          "us-south1",
+          "us-west1",
+          "us-west2",
+          "us-west3",
+          "us-west4"
+        ],
+        docs:
+          'https://cloud.google.com/compute/docs/regions-zones'
+      }),
+      TF_VAR_ZONE: validZone({ default: 'us-central1-a' }),
+      PEM_FILE_PATH: validCertPathStr({ default: '/home/ubuntu/ubuntu.pem' }),
+      TF_VAR_GCE_PUB_KEY_FILE: validStr({ default: '/home/ubuntu/aws-key.pem.pub' }),
+    })
+  }
+
   cleanEnv(process.env, {
```

Comment:
Don't we need all those removed checks anymore?

Reply:
We do removed the checks. Few checks are specific to AWS, so separated for them for GCP and AWS.